### DISCUSSION TITLE
Android Auto on ungoogled phones

### DISCUSSION THREAD

**Discussion by rtsisyk** [organicmaps/organicmaps#4778] [Q&A]
Folks, do you know whether it is possible to get Android Auto without Google Services? I have tried to install Android Auto APK from Avrora on my CalyxOS Android 13 device, but it doesn't work:

<img src="https://user-images.githubusercontent.com/1799054/224939548-1cad12bc-45d6-48cb-8a5e-91fbd32817c4.png" width="400px" >

<img src="https://user-images.githubusercontent.com/1799054/224939564-00b42d2c-9cba-4df5-9c3d-0801c1dfcf63.png" width="400px">

```
[W]: No configuration specified - using default values.
[E]: Could not load configuration from '~/.android/headunit.ini'.
BoringSSL is the SSL implementation used in the receiver-lib.
Starting link. Requested protocol version: 1.7
[I]: Searching for compatible USB device...
[I]: Found 8 USB devices.
[I]: Device 'Google Pixel 4a XXXXXXXX' (vid=18d1, pid=4ee7) supports AOAPv2, starting accessory mode...
[I]: Device 'XXXXXXXX' does not support AOAP, skipping.
[I]: Device 'Generic Mass Storage Device XXXXXXXX' does not support AOAP, skipping.
[I]: No device found ready yet, will retry shortly...
[I]: Found 8 USB devices.
[I]: Found device 'Google Pixel 4a XXXXXXXX' in accessory mode (vid=18d1, pid=2d01).
[I]: Found accessory: ifnum: 0, rd_ep: 129, wr_ep: 1
[I]: Attaching to USB device...
[I]: Attached!
```

I found that some people we able to get it running by flashing Android Auto APK into system partition:

- https://github.com/sn-00-x/aa4mg
- https://forum.xda-developers.com/t/microg-android-auto-fully-working.4319159/page-6

I have doubts that it is practically viable for the most users. Does anyone have ungoogled phone with working Android Auto?

**Comment by hb0nd** [comment]
looks like it's not really possible without a lot of configuration, even with microG: https://github.com/microg/GmsCore/issues/897

**Comment by daniandtheweb** [comment]
I'm able to use android auto on my ungoogled device using aa4mg magisk module and microg and it works flawlessly. In order to make it work you just have to install microg and then install aa4mg.
